core/src/main/java/org/jruby/ir/operands/Range.java

```diff
@@ -0,0 +1,82 @@
+package org.jruby.ir.operands;
+
+import org.jruby.RubyRange;
+import org.jruby.RubyRational;
+import org.jruby.ir.IRVisitor;
+import org.jruby.ir.persistence.IRReaderDecoder;
+import org.jruby.ir.persistence.IRWriterEncoder;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+/**
+ * Literal Range with literal endpoints.
+ */
+public class Range extends ImmutableLiteral {
+    private final ImmutableLiteral begin;
+    private final ImmutableLiteral end;
+    private final boolean exclusive;
+
+    public Range(ImmutableLiteral begin, ImmutableLiteral end, boolean exclusive) {
+        super();
+
+        this.begin = begin;
+        this.end = end;
+        this.exclusive = exclusive;
+    }
+
+    @Override
+    public OperandType getOperandType() {
+        return OperandType.RANGE;
```

Comment:
I don't see OperandType here?

Reply:
Oh it is already there...what! :)

Reply:
Ah ok this was originally an Operand and we realized since it could raise we had to make it buildRange.  I wonder why we never had the discussion of both since we can check to see if it will raise during IRBuild in cases?